crates/runtime/src/flight/get_flight_info.rs

```diff
@@ -0,0 +1,52 @@
+use arrow_flight::{
+    sql::{Any, Command},
+    FlightDescriptor, FlightEndpoint, FlightInfo, Ticket,
+};
+use prost::Message;
+use tonic::{Request, Response, Status};
+
+use super::{flightsql, to_tonic_err, Service};
+
+pub(crate) async fn handle(
+    flight_svc: &Service,
+    request: Request<FlightDescriptor>,
+) -> Result<Response<FlightInfo>, Status> {
+    let Ok(message) = Any::decode(&*request.get_ref().cmd) else {
```

Comment:
Is the desired behaviour, if we don't get any valid/known command, return a "raw" flight info response?

Reply:
Yeah, this is how we need to maintain compatibility with our current protocol.